lib/schemes/HttpUriPlugin.js

```diff
@@ -19,6 +19,37 @@ const memoize = require("../util/memoize");
 
 const getHttp = memoize(() => require("http"));
 const getHttps = memoize(() => require("https"));
+const proxyFetch = (request, proxy) => (url, options, callback) =>
+	Promise.resolve(
```

Comment:
This is very weird. A function with a `callback` that returns a `Promise`.

To me it looks like this can be implemented without Promises. Maybe return a EventEmitter for the error event instead?